src/network/connection.h

```diff
@@ -85,24 +178,58 @@ static inline float CALC_DTIME(u64 lasttime, u64 curtime)
 	return MYMAX(MYMIN(value,0.1),0.0);
 }
 
-struct BufferedPacket
-{
-	BufferedPacket(u8 *a_data, u32 a_size):
-		data(a_data, a_size)
-	{}
-	BufferedPacket(u32 a_size):
-		data(a_size)
-	{}
-	Buffer<u8> data; // Data of the packet, including headers
+/*
+	Struct for all kinds of packets. Includes following data:
+		BASE_HEADER
+		SharedBuffer<u8> packet data
+*/
+struct BufferedPacket {
+	BufferedPacket(u32 a_size)
+	{
+		m_data.resize(a_size);
+		data = &m_data[0];
+	}
+
+	DISABLE_CLASS_COPY(BufferedPacket)
+
+	void copyTo(BufferedPacket &p) const
+	{
+		p.m_data = m_data;
+		p.m_write_locked = m_write_locked;
+		p.data = data;
```

Comment:
should be `p.data = &p.m_data[0];`, otherwise it ends up pointing to the original object

is this method even used though?

Reply:
No, I thought it would be used during the PR writing process. However, there are no calls to this function, this removing. Good catch though. Thank you!